lib/DevServerManager.ts

```diff
@@ -1,48 +1,86 @@
-// @ts-nocheck
-const { logger } = require('@hubspot/local-dev-lib/logger');
-const { COMPONENT_TYPES } = require('./projects/structure');
-const { i18n } = require('./lang');
-const { promptUser } = require('./prompts/promptUtils');
-const { DevModeInterface } = require('@hubspot/ui-extensions-dev-server');
-const {
+import { logger } from '@hubspot/local-dev-lib/logger';
+import { Environment } from '@hubspot/local-dev-lib/types/Config';
+import {
+  COMPONENT_TYPES,
+  ComponentTypes,
+  Component,
+} from './projects/structure';
+import { i18n } from './lang';
+import { promptUser } from './prompts/promptUtils';
+import { DevModeInterface as UIEDevModeInterface } from '@hubspot/ui-extensions-dev-server';
+import {
   startPortManagerServer,
   stopPortManagerServer,
   requestPorts,
-} = require('@hubspot/local-dev-lib/portManager');
-const {
+} from '@hubspot/local-dev-lib/portManager';
+import {
   getHubSpotApiOrigin,
   getHubSpotWebsiteOrigin,
-} = require('@hubspot/local-dev-lib/urls');
-const { getAccountConfig } = require('@hubspot/local-dev-lib/config');
+} from '@hubspot/local-dev-lib/urls';
+import { getAccountConfig } from '@hubspot/local-dev-lib/config';
+import { ProjectConfig } from '../types/Projects';
 
 const i18nKey = 'lib.DevServerManager';
 
 const SERVER_KEYS = {
   privateApp: 'privateApp',
   publicApp: 'publicApp',
+} as const;
+
+type ServerKey = keyof typeof SERVER_KEYS;
+
+type DevServerInterface = {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-function-type
+  setup?: Function;
```

Comment:
Let me know what you think about this. eslint doesn't like the generic Function type, but I was having some trouble with this. The UIE dev server defines its own type (they call it SetupArguments), but they use their own type interfaces for it and they're slightly different from what we have defined on our end. So I was having trouble [here](https://github.com/HubSpot/hubspot-cli/pull/1335/files#diff-2c6c8dea63a8a0b77cc903f016f011e50bcc5d79c276fb59944f6cf63e34c27eR66). I just ignored the error for now, but I'm happy to see if I can get things working with a more specific type if we want.

Reply:
Messing around with this locally and I feel like there's probably a way to do this using generic types but i think either way will require some work in the UIE dev server repo so probably not worth it right now

Reply:
Ok, thanks for poking around and verifying. I'm going to leave this as is for now 👍